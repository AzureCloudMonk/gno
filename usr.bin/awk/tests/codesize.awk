# awk program to parse the output of "lseg -d <files>" and calculate
#    1. The total size of code in each object file
#    2. The name of any procedure that uses over "maxseg" bytes (default=100)

BEGIN {
	# If maxseg was not set on command line, use default
	if (maxseg <= 0)  maxseg = 100;
}

/ Code / {
	# Is this the beginning of a new file's statistics?
	if ($1 != filename)   {
		if (codesize > 0)   {
			sizes[filename] = codesize;
			}
		filename = $1;
		codesize = 0;
		}
	codesize += $3;
	# If there are > 4 fields, fourth one is the stack size
        if ($4 > maxseg) print $5," in file ", $1," uses ",$4," stack bytes"
}

END {
	# Print the report
        print "Code file         Code bytes"
	for (fname in sizes) {
        	printf "%s", fname;
                i = length(fname);
                while (i++ < 20) printf " ";
                printf " %6d\n", sizes[fname];
                }
}
