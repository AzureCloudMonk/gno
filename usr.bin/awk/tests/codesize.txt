File                 Type               Size     Stack  Name
-------------------- ------------------ -------- ------ ----------------
../main.o            Code                   3467     22 "main"
../main.o            Code                    669     14 "pgetc"
../main.o            Code                     38        "~GLOBALS"
../main.o            Code                     80        "~ARRAYS"
../b.o               Code                   1151     20 "makedfa"
../b.o               Code                    841     22 "mkdfa"
../b.o               Code                   1657     12 "makeinit"
../b.o               Code                    759      8 "penter"
../b.o               Code                    586      8 "freetr"
../b.o               Code                    377     16 "hexstr"
../b.o               Code                    506     16 "quoted"
../b.o               Code                   1151     24 "cclenter"
../b.o               Code                     89      2 "overflo"
../b.o               Code                   1400     14 "cfoll"
../b.o               Code                   1277     14 "first"
../b.o               Code                    333     12 "follow"
../b.o               Code                     87      8 "member"
../b.o               Code                    354     16 "match"
../b.o               Code                   1552     24 "pmatch"
../b.o               Code                   1525     24 "nematch"
../b.o               Code                    281     10 "reparse"
../b.o               Code                     43      6 "regexp"
../b.o               Code                    815     12 "primary"
../b.o               Code                    128      8 "concat"
../b.o               Code                     96      6 "alt"
../b.o               Code                    199      8 "unary"
../b.o               Code                   1270     20 "relex"
../b.o               Code                   3513     22 "cgoto"
../b.o               Code                   1123      8 "freefa"
../b.o               Code                     51        "~GLOBALS"
../b.o               Code                     80        "~ARRAYS"
../lib.o             Code                    537      6 "recinit"
../lib.o             Code                    392     58 "makefields"
../lib.o             Code                    182     12 "initgetrec"
../lib.o             Code                   1879     20 "getrec"
../lib.o             Code                     88        "nextfile"
../lib.o             Code                   1801     26 "readrec"
../lib.o             Code                    188     64 "getargv"
../lib.o             Code                    318     18 "setclvar"
../lib.o             Code                   3069     31 "fldbld"
../lib.o             Code                    240     12 "cleanfld"
../lib.o             Code                     97      6 "newfld"
../lib.o             Code                    180     10 "fieldadr"
../lib.o             Code                    214      8 "growfldtab"
../lib.o             Code                   1208     22 "refldbld"
../lib.o             Code                   1581     22 "recbld"
../lib.o             Code                    343      6 "yyerror"
../lib.o             Code                     73      2 "fpecatch"
../lib.o             Code                    112      4 "bracecheck"
../lib.o             Code                    363      6 "bcheck2"
../lib.o             Code                    838      6 "error"
../lib.o             Code                   3866     20 "eprint"
../lib.o             Code                    102      4 "bclass"
../lib.o             Code                    268     10 "errcheck"
../lib.o             Code                    303     12 "isclvar"
../lib.o             Code                    301     20 "isnumber"
../lib.o             Code                     46        "~GLOBALS"
../lib.o             Code                    450        "~ARRAYS"
../proctab.o         Code                    188     10 "tokname"
../proctab.o         Code                      1        "~GLOBALS"
../proctab.o         Code                   1380        "~ARRAYS"
../tran.o            Code                   1112      6 "syminit"
../tran.o            Code                    358     62 "arginit"
../tran.o            Code                    406     14 "envinit"
../tran.o            Code                    250     18 "makesymtab"
../tran.o            Code                    560     20 "freesymtab"
../tran.o            Code                    503     20 "freeelem"
../tran.o            Code                    830     16 "setsymtab"
../tran.o            Code                    144     12 "hash"
../tran.o            Code                    408     24 "rehash"
../tran.o            Code                    243     20 "lookup"
../tran.o            Code                    378     20 "setfval"
../tran.o            Code                    319      6 "funnyvar"
../tran.o            Code                    585     20 "setsval"
../tran.o            Code                    410     18 "getfval"
../tran.o            Code                    618    122 "getsval"
../tran.o            Code                    190     10 "tostring"
../tran.o            Code                   1031     24 "qstring"
../tran.o            Code                    105        "~GLOBALS"
../lex.o             Code                     37     10 "peek"
../lex.o             Code                   1379     24 "gettok"
../lex.o             Code                   3792     14 "yylex"
../lex.o             Code                   1919    128 "string"
../lex.o             Code                    203     16 "binsearch"
../lex.o             Code                    885     18 "word"
../lex.o             Code                      7        "startreg"
../lex.o             Code                    702     14 "regexpr"
../lex.o             Code                    303     10 "input"
../lex.o             Code                    320      6 "unput"
../lex.o             Code                    123      8 "unputstr"
../lex.o             Code                     43        "~GLOBALS"
../lex.o             Code                    991        "~ARRAYS"
../parse.o           Code                    209     14 "nodealloc"
../parse.o           Code                     44     10 "exptostat"
../parse.o           Code                    133     14 "node1"
../parse.o           Code                    200     14 "node2"
../parse.o           Code                    265     14 "node3"
../parse.o           Code                    332     14 "node4"
../parse.o           Code                     66     14 "stat1"
../parse.o           Code                     70     14 "stat2"
../parse.o           Code                     74     14 "stat3"
../parse.o           Code                     78     14 "stat4"
../parse.o           Code                     66     14 "op1"
../parse.o           Code                     70     14 "op2"
../parse.o           Code                     74     14 "op3"
../parse.o           Code                     78     14 "op4"
../parse.o           Code                    101     14 "celltonode"
../parse.o           Code                     51      6 "rectonode"
../parse.o           Code                    344     14 "makearr"
../parse.o           Code                    172     14 "pa2stat"
../parse.o           Code                    231     14 "linkum"
../parse.o           Code                    283     12 "defn"
../parse.o           Code                    209     14 "isarg"
../parse.o           Code                      3        "~GLOBALS"
../parse.o           Code                    100        "~ARRAYS"
../run.o             Code                    353     16 "adjbuf"
../run.o             Code                     31      2 "run"
../run.o             Code                    854     22 "execute"
../run.o             Code                    940     14 "program"
../run.o             Code                   2949     54 "call"
../run.o             Code                    238     14 "copycell"
../run.o             Code                    351     10 "arg"
../run.o             Code                    881     20 "jump"
../run.o             Code                   1323     32 "getline"
../run.o             Code                     91     10 "getnf"
../run.o             Code                   1085     42 "array"
../run.o             Code                    944     38 "adelete"
../run.o             Code                   1073     42 "intest"
../run.o             Code                    821     40 "matchop"
../run.o             Code                    587     22 "boolop"
../run.o             Code                    805     30 "relop"
../run.o             Code                    306      6 "tfree"
../run.o             Code                    387     16 "gettemp"
../run.o             Code                    327     20 "indirect"
../run.o             Code                   1171     34 "substr"
../run.o             Code                    550     50 "sindex"
../run.o             Code                   3191     44 "format"
../run.o             Code                    482     24 "awksprintf"
../run.o             Code                    911     30 "awkprintf"
../run.o             Code                   1335     48 "arith"
../run.o             Code                    285     22 "ipow"
../run.o             Code                    425     28 "incrdecr"
../run.o             Code                   1466     48 "assign"
../run.o             Code                    530     30 "cat"
../run.o             Code                    321     14 "pastat"
../run.o             Code                    558     16 "dopa2"
../run.o             Code                   3051    101 "split"
../run.o             Code                    283     14 "condexpr"
../run.o             Code                    354     14 "ifstat"
../run.o             Code                    392     14 "whilestat"
../run.o             Code                    412     14 "dostat"
../run.o             Code                    616     14 "forstat"
../run.o             Code                    704     36 "instat"
../run.o             Code                   1848     46 "bltin"
../run.o             Code                    670     22 "printstat"
../run.o             Code                     65      6 "nullproc"
../run.o             Code                    209     22 "redirect"
../run.o             Code                   1458     22 "openfile"
../run.o             Code                    206     12 "filename"
../run.o             Code                   1262     18 "closefile"
../run.o             Code                    686     10 "closeall"
../run.o             Code                   1890     48 "sub"
../run.o             Code                   2844     54 "gsub"
../run.o             Code                    706     14 "backsub"
../run.o             Code                     55        "~GLOBALS"
../run.o             Code                    689        "~ARRAYS"
../ytab.o            Code                     16      4 "yywrap"
../ytab.o            Code                    214      6 "setfname"
../ytab.o            Code                    134      8 "constnode"
../ytab.o            Code                    108     10 "strnode"
../ytab.o            Code                    140     12 "notnull"
../ytab.o            Code                    222     10 "checkdup"
../ytab.o            Code                  17185     32 "yyparse"
../ytab.o            Code                     45        "~GLOBALS"
../ytab.o            Code                  13274        "~ARRAYS"
