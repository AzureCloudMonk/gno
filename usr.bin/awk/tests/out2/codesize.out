"getsval"  in file  ../tran.o  uses  122  stack bytes
"string"  in file  ../lex.o  uses  128  stack bytes
"split"  in file  ../run.o  uses  101  stack bytes
Code file         Code bytes
../main.o              4254
../b.o                21244
../lex.o              10704
../parse.o             3253
../run.o              43971
../lib.o              19039
../proctab.o           1569
../tran.o              8450
